### DISCUSSION TITLE
Change keys of self hosted server

### DISCUSSION THREAD

**Discussion by mpgamelas** [rustdesk/rustdesk#2567] [Q&A] [answered]
When reinstalling the server on a new machine, I would like to keep the same keys from the old installation. However, when changing the keys in both files, the connection works, but it says it's non-encrypted.

I followed the self host guide without docker:
https://rustdesk.com/docs/en/self-host/install/

Thanks in advance for the help.

**Comment by rustdesk** [comment, marked as answer]
Copy id_ed25519 and id_ed25519.pub files to your new server, then restart hbbs/hbbr.

**Reply by mpgamelas** [reply]
Thanks for the help. I will try this on a later date to see if it stills says it's non encrypted.

Restarting hbbs/hbbr is just executing then directly on the terminal or is there some systemctl command to do so?

**Comment by elchmartin** [comment]
I kind of have the same problem, my vServer lost all/any network connection, so I cannot copy files from it.
But i can still use a VNC connection, so i can display the contents of the files.
Can I just type the keys into the files on my new installation? Will it give me an error/not encrypted message?

**Reply by xlionjuan** [reply]
RustDesk server is using **Asymmetric cryptography**, so the server needs **private key** to work.
